src/network/networkd-conf.c

```diff
@@ -152,4 +161,11 @@ int config_parse_duid_rawdata(
         *dhcp_duid_len = duid_index;
 
         return 0;
+
+fail:
+        /* Ignore invalid DUID assignment */
+        *duid_type = _DUID_TYPE_INVALID;
+        *dhcp_duid_len = 0;
+        *dhcp_duid_type = 0;
+        return 0;
```

Comment:
The general rule in systemd (99.9% of cases) is "don't modify anything on failure, let the caller deal with that". Having things behave consistently in this way really helps avoid surprising side effects.